src/trex/quality_control.py

```diff
@@ -0,0 +1,375 @@
+
+def load_reads(data_dir: pathlib.Path) -> pd.DataFrame:
+    """Loads saved reads into a DataFrame"""
+    READ_DIR = data_dir / 'reads.txt'
+    return pd.read_csv(READ_DIR, delimiter='\t')
+
+
+def load_molecules(data_dir: pathlib.Path) -> pd.DataFrame:
+    """Loads saved molecules before correcting into a DataFrame."""
+    MOLS_DIR = data_dir / 'molecules.txt'
+    return pd.read_csv(MOLS_DIR, delimiter='\t')
+
+
+def load_molecules_corrected(data_dir: pathlib.Path) -> pd.DataFrame:
+    """Loads saved molecules after correcting into a DataFrame."""
+    MOLS_DIR = data_dir / 'molecules_corrected.txt'
+    return pd.read_csv(MOLS_DIR, delimiter='\t')
+
+
+def load_cells(data_dir: pathlib.Path,
+               filtered: bool = True) -> pd.DataFrame:
+    cells_df = []
+
+    filename = 'cells_filtered.txt' if filtered else 'cells.txt'
+
+    with open(data_dir / filename) as file:
+        next(file)
+        while line := file.readline():
+            cell_id, clone_id_info = line[:-1].split('\t:\t')
+            clone_id_info = clone_id_info.split('\t')
+
+            clone_ids_df = {'clone_id': clone_id_info[::2],
+                           'counts': clone_id_info[1::2]}
+            clone_ids_df = pd.DataFrame(clone_ids_df,
+                                       index=np.arange(len(clone_id_info[::2])))
+            clone_ids_df['cell_id'] = cell_id
+            cells_df.append(clone_ids_df)
+
+    cells_df = pd.concat(cells_df, ignore_index=True)
+    cells_df['clone_id'] = cells_df.clone_id.astype('category')
+    cells_df['counts'] = cells_df.counts.astype(int)
+    cells_df['cell_id'] = cells_df.cell_id.astype('category')
+
+    return cells_df
+
+
+def load_umi_count_matrix(data_dir: pathlib.Path):
+    """Loads saved UMI count matrix into a DataFrame."""
+    UMI_DIR = data_dir / 'umi_count_matrix.csv'
+    return pd.read_csv(UMI_DIR)
+
+
+def load_clone_ids(data_dir: pathlib.Path):
+    """Loads saved clone id and cell id into a DataFrame."""
+    return pd.read_csv(data_dir / 'clones.txt')
+
+
+def read_quality(reads: pd.DataFrame,
+                 ax: plt.Axes = None,
+                 add_description: bool = True) -> plt.Axes:
+    """Plot histogram of how many times a molecule was read."""
+    count_reads = reads.groupby(['#cell_id', 'umi']).agg('count')
+
+    ax = sns.histplot(data=count_reads, x='clone_id', discrete=True, log=True,
+                      ax=ax)
+    plt.xlabel('Number of reads')
+    plt.title('Number of reads per molecule')
+
+    if add_description:
+        txt = 'This plot shows how many times a molecule has been read. \n' \
+              'The more reads per molecule, the better.'
+        plt.text(0, -0.3, txt, transform=ax.transAxes, size=12)
+    return ax
+
+
+def get_read_per_molecule(df: pd.DataFrame) -> pd.Series:
+    """Get a pandas Series with the number of reads per molecule."""
+    return df.groupby(['#cell_id', 'umi']).clone_id.agg('count')
+
+
+def get_length_read(df: pd.DataFrame,
```

Comment:
The word "read" is ambiguous and is used in two different meanings: It can be the noun meaning "sequencing read" (pronounced "reed") or the past participle of "to read" (pronounced "red").

Ways to disambiguate:
- Use plural for the noun: `get_reads_per_molecule`
- Avoid the word: `get_nucleotides_per_molecule`